src/scf/self_consistent_field.jl

```diff
@@ -35,6 +35,27 @@ function scf_default_callback(info)
     @printf "%3d    %-15.12f    %E\n" neval E res
 end
 
+"""
+Flag convergence as soon as total energy change drops below tolerance
+"""
+function scf_convergence_energy_difference(tolerance)
+    energy_total = NaN
+
+    function is_converged(info)
+        etot_old = energy_total
+        energy_total = sum(values(info.energies))
+        abs(energy_total - etot_old) < tolerance
+    end
+    return is_converged
```

Comment:
That closure trick is neat. It's a bit brittle, because it assumes that it'll only get called with "proper iterates" (as opposed to eg line searches). But it's fine for now (and we already assume that at other places anyway)